src/store/chart_modifiers/line_bar/base.js

```diff
@@ -51,20 +51,46 @@ export default class LineBaseModifier {
       case 'day':
         delta = 86400
         break
+      case 'month':
+        monthDays = (new Date(startDate.getFullYear(), startDate.getMonth() + 1, 0)).getDate()
+        delta = 60 * 60 * 24 * monthDays
+        break
     }
     delta *= payload.dateInterval
-
-    // set the offset if there is one we need to account for
-    const offset = (payload.timeZoneOffset) ? payload.timeZoneOffset : 0
+    payload.dateStart = payload.dateStart - (payload.dateStart % 900)
+    payload.dateEnd = payload.dateEnd - (payload.dateEnd % 900)
 
     for (let i = payload.dateStart; i <= payload.dateEnd; i += delta) {
+      let oldDate = (new Date(i * 1000))
+      if (payload.intervalUnit === 'month') {
+        let monthDaysCurrent = (new Date(oldDate.getFullYear(), oldDate.getMonth() + 1, 0)).getDate()
+
+        delta += (monthDaysCurrent - monthDays) * 24 * 60 * 60
+        monthDays = monthDaysCurrent
+      }
+      let dataDate = (new Date((i + delta) * 1000))
       try {
         let accumulator = 0
         if (isNaN(resultDataObject.get(i + delta)) || isNaN(resultDataObject.get(i))) {
           continue
         }
-        accumulator = resultDataObject.get(i + delta)
-        returnData.push({ x: (new Date((i + delta + offset) * 1000)), y: accumulator })
+        if (Math.abs(resultDataObject.get(i + delta)) < Math.abs(resultDataObject.get(i))) {
+          continue
+        }
+        // If either reading is zero that indicates a missing reading -- do not report.
+        if (resultDataObject.get(i + delta) === 0 || resultDataObject.get(i) === 0) {
+          continue
+        }
+        accumulator = resultDataObject.get(i + delta) - resultDataObject.get(i)
```

Comment:
This particular step is specific for accumulated_real to get the energy usage in an interval.  We don't want to perform this math on real_power or other meter metrics like amperage or current as far as I can recall.